src/dashboardGenerator.js

```diff
@@ -31,23 +31,25 @@ const SEVEN_DAYS = 60 * 60 * 24 * 7;
 
 const applyLimits = state => {
   // See: https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_limits.html
-  const maxMetricsPerWidget = 100;
-  const maxMetricsPerServiceInAWidget = 2;
-  const maxServices = Math.floor(maxMetricsPerWidget / maxMetricsPerServiceInAWidget);
+  const maxMetricsPerDashboard = 500;
+  const maxMetricsPerServiceInAWidget = 1;
+  const maxWidget = 6
+  const maxServices = Math.floor(maxMetricsPerDashboard / (maxWidget * maxMetricsPerServiceInAWidget));
 
   // For each service there will be three metrics (MTTR, MTBF, MTTF)
-  const maxSupportedMetrics = maxServices * 3;
-
-  if (state.metrics.length > maxSupportedMetrics) {
-    console.warn(
-      `Maximum of ${maxSupportedMetrics} metrics are allowed in a single dashboard. Some metrics will not be reported.`,
-    );
-    console.log('List of Metrics before truncate: ', JSON.stringify(state.metrics));
-    state.metrics = state.metrics.sort((a, b) => (a.serviceName > b.serviceName ? 1 : -1));
-    state.metrics = state.metrics.slice(0, maxSupportedMetrics);
-    console.log('List of Metrics after truncate: ', JSON.stringify(state.metrics));
-    state.expectTruncated = true;
-    state.yOffset = 2;
+  const maxSupportedMetrics = maxServices ;
+
+  for(let i = 0; i<state.widgetMappings.length; i++) {
+    state.widgetMappings[i].filteredMetrics = state.metrics.filter(x => x.metricName === state.widgetMappings[i].label);
+    if (state.widgetMappings[i].filteredMetrics.length > maxSupportedMetrics) {
+      console.warn(
+        `Maximum of ${maxSupportedMetrics} metrics are allowed in a single dashboard. Some metrics will not be reported.`,
+      );
+      state.widgetMappings[i].filteredMetrics = state.widgetMappings[i].filteredMetrics.sort((a, b) => (a.serviceName > b.serviceName ? 1 : -1));
```

Comment:
this is filtering alphabetically?
Ascending/descending?

Happy with the approach - but it would be better to encapsulate the logic in an appropriately named method

e.g. `truncateAtoZ` or `truncateZtoA`